sled-hardware/src/illumos/partitions.rs

```diff
@@ -119,6 +191,11 @@ fn internal_ensure_partition_layout<GPT: gpt::LibEfiGpt>(
             };
             match variant {
                 DiskVariant::U2 => {
+                    // First we need to check that this disk is of the proper
+                    // size and correct logical block address formatting.
+                    ensure_size_and_formatting(log, identity)?;
```

Comment:
Mostly for my own education: what does this do with disks that already have data on them?

Reply:
We don't touch a disk with a label/zpool as changing the LBA format erases all data on the disk. This should only be preformed when StorageManger finds a new disk that has not been setup yet i.e first boot, add disk etc

Reply:
And I expect that when we transform to an explicit adoption model then we'll want this to not occur until that adoption process is initiated.

Reply:
This only gets run if there is no GPT table. It's hidden by github above the diff.